src/main/java/wromaciej/hvac_sim/thermo/matter/fluids/service/FluidData.java

```diff
@@ -1,22 +1,39 @@
 package wromaciej.hvac_sim.thermo.matter.fluids.service;
 
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
 import thermoCP.CoolProp;
+import wromaciej.hvac_sim.thermo.matter.fluids.model.FluidName;
 import wromaciej.hvac_sim.thermo.matter.fluids.parameters.*;
 import wromaciej.hvac_sim.thermo.quantities.base.AirQuantity;
 import wromaciej.hvac_sim.thermo.unitSystems.UnitSystem;
 
 /**
  * ThermoCP library ADAPTER
  */
+
+@Service
 public class FluidData {
 
-    public void loadLibrary(){
+    private boolean isLibraryLoaded = false;
+
+    public boolean isLibraryLoaded(){
+        return isLibraryLoaded;
+    }
+
+    public FluidData(@Value("true") boolean shouldLoadLibrary) {
+        if ((!isLibraryLoaded) && (shouldLoadLibrary)) loadLibrary();
+    }
+
+    private void loadLibrary(){
+        isLibraryLoaded = false;
         try{
```

Comment:
I've already posted a comment that it should not be done like that. find the comment